pkg/connect/state/state.go

```diff
@@ -254,28 +272,27 @@ func (g *WorkerGroup) Sync(ctx context.Context, groupManager WorkerGroupManager,
 
 			// Wait for other sync to complete and retry
 			if perr.Code == syscode.CodeSyncAlreadyPending {
-				select {
-				case <-ctx.Done():
-					return fmt.Errorf("could not wait for sync to complete")
-				case <-time.After(2 * time.Second):
-				}
-				attempt++
 				continue
 			}
 		}
 
-		break
-	}
+		if err := json.NewDecoder(resp.Body).Decode(&syncReply); err != nil {
+			if errors.Is(err, io.EOF) {
```

Comment:
If the response was successful but we got an EOF, retry (we will get the same response as the syncing endpoint is idempotent)